app/celery/tasks.py

```diff
@@ -287,58 +297,57 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Any],
 
     check_service_over_daily_message_limit(KEY_TYPE_NORMAL, service)
 
-    research_mode = service.research_mode  # type: ignore
-
     current_app.logger.info(f"Sending following sms notifications to AWS: {notification_id_queue.keys()}")
-    for notification in saved_notifications:
-        queue = notification_id_queue.get(notification.id) or template.queue_to_use()  # type: ignore
+    for notification_obj in saved_notifications:
+        check_if_request_would_put_service_over_daily_sms_limit(KEY_TYPE_NORMAL, service, notification_obj.billable_units)  # type: ignore
+
+        queue = notification_id_queue.get(notification_obj.id) or template.queue_to_use()  # type: ignore
         send_notification_to_queue(
-            notification,
-            research_mode,
+            notification_obj,
+            service.research_mode,
             queue=queue,
         )
 
         current_app.logger.debug(
             "SMS {} created at {} for job {}".format(
-                notification.id,
-                notification.created_at,
-                notification.job,
+                notification_obj.id,
+                notification_obj.created_at,
+                notification_obj.job,
             )
         )
 
 
 @notify_celery.task(bind=True, name="save-emails", max_retries=5, default_retry_delay=300)
 @statsd(namespace="tasks")
-def save_emails(self, service_id: Optional[str], signed_notifications: List[Any], receipt: Optional[UUID]):
+def save_emails(self, _service_id: Optional[str], signed_notifications: List[Any], receipt: Optional[UUID]):
```

Comment:
why did we rename this variable?

Reply:
Renamed because below on line 338, `service_id` is set to a different value. It goes from being `Optional[str]` to `str`, i.e. it can't be `None` anymore, since it is used to look up the service from the db. The type checker mypy fails the check if a variable changes type. 

But I probably could have chosen a better name here!

Reply:
I think mypy enforces the rule that variables shouldn't change types because it makes the code easier to understand